Automattic/sensei
# Cleanup student progress when course/lesson/quiz/user deleted

Fixes #6060

When the admin deletes a course, or a lesson, or a quiz, or a user, we have to delete orphan progress. It makes no sense to keep it in DB, as it doesn't contain sensible information.

### Changes proposed in this Pull Request

* Cleanup student progress when a course/lesson/quiz/user deleted

### Testing instructions

* Add a couple of courses with lessons and quizzes.
* Add a student user (you'll need to delete it later).
* Start both courses and complete lessons.
* Start with deleting lessons: check the corresponding progress disappeared.
* Delete one course: check the corresponding progress disappeared.
* Delete the student: check all their progress disappeared.
